src/server/scripts/Northrend/VaultOfArchavon/boss_toravon.cpp

```diff
@@ -41,122 +48,136 @@ enum Misc
+        }
 
-            void EnterCombat(Unit* /*who*/)
+        void JustDied(Unit*)
+        {
+            if (pInstance)
             {
-                me->CastSpell(me, SPELL_FROZEN_MALLET, true);
-
-                events.ScheduleEvent(EVENT_FROZEN_ORB_STALKER, 12000);
-                events.ScheduleEvent(EVENT_FREEZING_GROUND, 7000);
-                if (pInstance)
-                    pInstance->SetData(EVENT_TORAVON, IN_PROGRESS);
+                pInstance->SetData(EVENT_TORAVON, DONE);
+                pInstance->DoRemoveAurasDueToSpellOnPlayers(SPELL_WHITEOUT);
             }
+        }
 
-            void JustDied(Unit* )
-            {
-                if (pInstance)
-                {
-                    pInstance->SetData(EVENT_TORAVON, DONE);
-                    pInstance->DoRemoveAurasDueToSpellOnPlayers(SPELL_WHITEOUT);
-                }
-            }
+        void JustSummoned(Creature* cr)
+        {
+            summons.Summon(cr);
+        }
 
-            void JustSummoned(Creature* cr)
-            {
-                summons.Summon(cr);
-            }
+        void UpdateAI(uint32 diff)
+        {
 
-            void UpdateAI(uint32 diff)
-            {
-                if (!UpdateVictim())
-                    return;
 
-                events.Update(diff);
-                if (me->HasUnitState(UNIT_STATE_CASTING))
-                    return;
+            if (!UpdateVictim())
+                return;
 
-                switch (events.GetEvent())
+            events.Update(diff);
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+
+            switch (events.GetEvent())
+            {
+            case EVENT_FREEZING_GROUND:
+                if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
+                me->CastSpell(target, SPELL_FREEZING_GROUND, false);
+                events.RepeatEvent(20000);
+                break;
+            case EVENT_FROZEN_ORB_STALKER:
+                me->CastCustomSpell(SPELL_FROZEN_ORB, SPELLVALUE_MAX_TARGETS, RAID_MODE(1, 3), me, false);
+                events.RepeatEvent(30000);
+                break;
+            case EVENT_CAST_WHITEOUT:
+                me->CastSpell(me, SPELL_WHITEOUT, false);
+                events.ScheduleEvent(EVENT_CAST_WHITEOUT_GROUND_EFFECT, 1000); // triggers after 1 sec "plus 1 from trigger to cast visual"
+                events.RepeatEvent(40000); // next whiteout instead first 25 SEC is now 45 SEC 
+                break;
+            case EVENT_CAST_WHITEOUT_GROUND_EFFECT: // Whiteout Ground effect trigger
+                if (Unit* whiteOutGround = me->SummonCreature(NPC_WHITEOUT_GROUND_EFFECT, -43.3316, -288.708, 92.2511, 1.58825, TEMPSUMMON_TIMED_DESPAWN, 4000))
+                whiteOutGround->CastSpell(whiteOutGround, SPELL_WHITEOUT_VISUAL, false); // Cast the spell
+                events.PopEvent();
+                break;
+            case EVENT_CHECK_SUMMONS:
```

Comment:
Where start this event ? you delete events.ScheduleEvent(EVENT_CHECK_SUMMONS, 10000);

Reply:
@arthaswow

Reply:
> @arthaswow

remove label To Be Merged

Reply:
?

Reply:
> ?

Where start this event ?

Reply:
Removed useless event, orbs doesn't need to be "despawned"